assets/src/editor/events/hooks/use-event-venue-entity.js

```diff
@@ -0,0 +1,55 @@
+import { useSelect } from '@wordpress/data';
+import { isModelEntityOfModel } from '@eventespresso/validators';
+
+const DEFAULT_OBJECT = {
+	venueEntity: null,
+	venueEntityLoaded: false,
+};
+
+/**
+ * @typedef { Object }         VenueResponse
+ * @property {BaseEntity|null} venueEntity       The instance of the venue or
+ *                                               null.
+ * @property {boolean}         venueEntityLoaded Whether the venue has been
+ *                                               loaded or not.
+ */
+
+/**
+ * A custom hook for retrieving the venue related to the given event
+ *
+ * @param {BaseEntity} eventEntity  An instance of an event entity.
+ *
+ * @return {VenueResponse} An object containing details about the retrieved
+ *                         related venue to the event.
+ */
+const useEventVenueEntity = ( eventEntity ) => {
+	const { venueEntity, venueEntityLoaded } = useSelect( ( select ) => {
+		const { getRelatedEntities, hasFinishedResolution } = select(
+			'eventespresso/core'
+		);
+		if ( isModelEntityOfModel( eventEntity, 'event' ) ) {
```

Comment:
what about displaying a warning if the incoming event is invalid to help with troubleshooting?
as is this would just return the default object with no indication of why the venue isn't loading

Reply:
Ya good point, will do.